### PR TITLE
add sha256 files and update README

### PR DESCRIPTION
**Pull request by johnk-novu**
##### Checklist

- [x] documentation is changed or added (if applicable)

### Description of change

Generate and publish sha256 of each "release" file. Update README for usage.

I am not sure if the travis tests will work, or how the files would get published to GitHub releases exactly.

Issue: https://github.com/aelsabbahy/goss/issues/447

~TODO: I still need to get the dgoss.sha256 generated in the Makefile.~

If https://github.com/aelsabbahy/goss/pull/567 is merged first, I will resolve the resulting merge conflicts (and vice-versa if this one is merged first).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,12 +47,16 @@ curl -fsSL https://goss.rocks/install | GOSS_VER=v0.3.6 GOSS_DST=~/bin sh
 
 ```bash
 # See https://github.com/aelsabbahy/goss/releases for release versions
-curl -L https://github.com/aelsabbahy/goss/releases/download/_VERSION_/goss-linux-amd64 -o /usr/local/bin/goss
-chmod +rx /usr/local/bin/goss
+curl -L https://github.com/aelsabbahy/goss/releases/download/_VERSION_/goss-linux-amd64 -o ./goss-linux-amd64
```
Comment (aelsabbahy) [inline README.md:50]: I would drop the documentation changes.

The installation instructions are intended to be quick/easy ones showing how simple goes is. If a user wants to take it further with checksums or installing somewhere other than /use/local/bin that's up to them, but I don't think that should be in the readme.

Comment (johnk-novu) [reply README.md:50]: 👍

**Comment by aelsabbahy** [comment]
Wonder if this hit the travis outage. I don't see a build for this.

I'm going to go ahead and click the "update branch" button in the hopes that it triggers travis.

**Comment by aelsabbahy** [comment]
Cool, that worked. The correct files seem to be generated in travis. Hopefully the next release has them

**Comment by aelsabbahy** [comment]
Thank you for your contribution

**Pull request merged**